ngmt/modules/td/_pham.py

```diff
@@ -0,0 +1,372 @@
+    occur within a turn. Hesitations are marked based on specific conditions related to the magnitude and 
+    continuity of the yaw angle changes.
+
+    Then, the detected turns are characterized by their start and end times, duration, angle of turn, peak 
+    angular velocity, and direction (left or right). Turns with angles equal to or greater than 90 degrees 
+    and durations between 0.5 and 10 seconds are selected for further analysis. Finally, the detected turns 
+    along with their characteristics (onset, duration, direction, etc.) are stored in a pandas DataFrame 
+    (detected_turns attribute).
+
+    Optionally, if plot_results is set to True, the algorithm generates a plot visualizing the accelerometer 
+    and gyroscope data alongside the detected turns. This visualization aids in the qualitative assessment of 
+    the algorithm's performance and provides insights into the dynamics of the detected turns.
+
+    Attributes:
+        gyro_convert_unit (float): Conevrsion of gyro data unit from deg/s to rad/s.
+        tracking_systems (str, optional): Tracking systems used. Default is 'imu'.
+        tracked_points (str, optional): Tracked points on the body. Default is 'LowerBack'.
+
+    Methods:
+        detect(data, sampling_freq_Hz):
+            Detects turns using accelerometer and gyro signals.
+
+            Args:
+                data (pd.DataFrame): Input accelerometer and gyro data (N, 6) for x, y, and z axes.
+                acceleration_data_unit (str): Unit of acceleration data.
+                gyro_data_unit (str): Unit of gyro data.
+                sampling_freq_Hz (float, int): Sampling frequency of the signals in Hz.
+                plot_results (bool, optional): If True, generates a plot. Default is False.
+
+            Returns:
+                PhamTurnDetection: an instance of the class with the detected turns
+                stored in the 'detected_turns' attribute.
+
+    Examples:
+        >>> pham = PhamTurnDetection()
+        >>> pham.detect(
+                data=input_data,
+                acceleration_data_unit='g'.
+                gyro_data_unit='deg/s'.
+                sampling_freq_Hz=200.0,
+                )
+        >>> print(pham.detected_turns)
+                onset   duration   event_type   direction_of_turn   angle of turn (deg)   peak angular velocity (deg/s)   tracking_systems    tracked_points
+            0   4.04    3.26       turn         left               -197.55                159.45                          imu                 LowerBack
+            1   9.44    3.35       turn         right               199.69                144.67                          imu                 LowerBack
+
+    References:
+        [1] Pham et al. (2017). Algorithm for Turning Detection and Analysis Validated under Home-Like Conditions...
+    """
+
+    def __init__(
+        self,
+        gyro_convert_unit: float = np.pi/180,
```

Comment:
What is this parameters? Do we need this, or can we use `numpy` built-in function, or simply do the multiplication inline?

Reply:
It is used to convert gyro data from deg/s to rad/s. This is because some calculations and functions may require the gyro data in radians instead of degrees.